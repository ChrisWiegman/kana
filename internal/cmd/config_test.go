package cmd

import (
	"bytes"
	"os/exec"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestConfig(t *testing.T) {
	t.Run("Test the default config command", func(t *testing.T) {
		cmd := exec.Command("../../build/kana", "config")
		var out bytes.Buffer
		cmd.Stdout = &out
		err := cmd.Run()
		if err != nil {
			t.Fatalf("Unexpected error: %v", err)
		}

		snaps.MatchSnapshot(t, out.String())
	})

	t.Run("Retrieve the PHP value from the config command", func(t *testing.T) {
		cmd := exec.Command("../../build/kana", "config", "php")
		var out bytes.Buffer
		cmd.Stdout = &out
		err := cmd.Run()
		if err != nil {
			t.Fatalf("Unexpected error: %v", err)
		}

		snaps.MatchSnapshot(t, out.String())
	})
}
