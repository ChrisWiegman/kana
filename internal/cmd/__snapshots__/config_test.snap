
[TestConfig/Test_the_default_config_command - 1]
┌──────────────────────┬─────────────────────┬─────────────────────┐
│       Setting        │    Global Value     │     Local Value     │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ activate             │ [1mtrue[0m                │ [1mtrue[0m                │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ admin.email          │ [1madmin@sites.kana.li[0m │ [1madmin@sites.kana.li[0m │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ admin.password       │ [1mpassword[0m            │ [1mpassword[0m            │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ admnin.username      │ [1madmin[0m               │ [1madmin[0m               │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ automaticLogin       │ [1mtrue[0m                │ [1mtrue[0m                │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ databaseClient       │ [1mphpmyadmin[0m          │ [1mphpmyadmin[0m          │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ environment          │ [1mlocal[0m               │ [1mlocal[0m               │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ imageUpdateDays      │ [1m7[0m                   │                     │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ mailpit              │ [1mfalse[0m               │ [1mfalse[0m               │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ mariadb              │ [1m11[0m                  │ [1m11[0m                  │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ multisite            │ [1mnone[0m                │ [1mnone[0m                │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ php                  │ [1m8.2[0m                 │ [1m8.2[0m                 │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ plugins              │                     │                     │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ removeDefaultPlugins │ [1mfalse[0m               │ [1mfalse[0m               │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ ssl                  │ [1mfalse[0m               │ [1mfalse[0m               │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ scriptdebug          │ [1mfalse[0m               │ [1mfalse[0m               │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ theme                │                     │                     │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ type                 │ [1msite[0m                │ [1msite[0m                │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ wpdebug              │ [1mfalse[0m               │ [1mfalse[0m               │
├──────────────────────┼─────────────────────┼─────────────────────┤
│ xdebug               │ [1mfalse[0m               │ [1mfalse[0m               │
└──────────────────────┴─────────────────────┴─────────────────────┘

---

[TestConfig/Retrieve_the_PHP_value_from_the_config_command - 1]
8.2

---
